src/Doctrine/Orm/Filter/IriSearchFilter.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Filter;
+
+use ApiPlatform\Doctrine\Common\Filter\IriSearchFilterTrait;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\OpenApiParameterFilterInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\ParameterProviderFilterInterface;
+use ApiPlatform\Metadata\PropertiesFilterInterface;
+use ApiPlatform\State\Provider\IriConverterParameterProvider;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class IriSearchFilter extends AbstractFilter implements OpenApiParameterFilterInterface, PropertiesFilterInterface, ParameterProviderFilterInterface
```

Comment:
I'd suggest not to extend the AbstractFilter, there's quite some work there that we don't need anymore using a `Parameter`. You may implement The `ApiPlatform\Doctrine\Orm\Filter\FilterInterfface` like this:

```
    public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void;
    
    public function getDescription(string $resourceClass): array {
        throw new RuntimeException('Not implemented');
    }

```